libraries/AP_Scripting/examples/ahrs-source.lua

```diff
@@ -0,0 +1,177 @@
+  -- auto source from RCx_FUNCTION = 83 (ZigZag_Auto)
+  local rc_function_source = rc:find_channel_for_option(90)
+  local rc_function_auto = rc:find_channel_for_option(83)
+  if (rc_function_source == nil) or (rc_function_auto == nil) then
+    gcs:send_text(0, "ahrs-source.lua: RCx_FUNCTION=90 or 83 not set!")
+    return update, 1000
+  end
+
+  -- check rangefinder distance threshold has been set
+  local rangefinder_thresh_dist = param:get('SCR_USER1')  -- SCR_USER1 holds rangefinder threshold
+  if (rangefinder_thresh_dist == nil) or (rangefinder_thresh_dist <= 0) then
+    gcs:send_text(0, "ahrs-source.lua: set SCR_USER1 to rangefinder threshold")
+    return update, 1000
+  end
+
+  -- check GPS speed accuracy threshold has been set
+  local gps_speedaccuracy_thresh = param:get('SCR_USER2')  -- SCR_USER2 holds GPS speed accuracy threshold
+  if (gps_speedaccuracy_thresh == nil) or (gps_speedaccuracy_thresh <= 0) then
+    gcs:send_text(0, "ahrs-source.lua: set SCR_USER2 to GPS speed accuracy threshold")
+    return update, 1000
+  end
+
+  -- check external nav innovation threshold has been set
+  local extnav_innov_thresh = param:get('SCR_USER3')  -- SCR_USER3 holds Non-GPS vertical velocity innovation
+  if (extnav_innov_thresh == nil) or (extnav_innov_thresh <= 0) then
+    gcs:send_text(0, "ahrs-source.lua: set SCR_USER3 to ExtNav innovation threshold")
+    return update, 1000
+  end
+
+  -- check if GPS speed accuracy is over threshold
+  local gps_speed_accuracy = gps:speed_accuracy(0)
+  local gps_over_threshold = (gps_speed_accuracy == nil) or (gps:speed_accuracy(0) > gps_speedaccuracy_thresh)
+  local gps_usable = (gps_speed_accuracy ~= nil) and (gps_speed_accuracy <= gps_usable_accuracy)
+
+  -- get external nav innovations from ahrs
+  local extnav_innov = Vector3f()
+  local extnav_var = Vector3f()
+  extnav_innov, extnav_var = ahrs:get_vel_innovations_and_variances_by_source(6)
+  local extnav_over_threshold = (extnav_innov == nil) or (extnav_innov:z() == 0.0) or (math.abs(extnav_innov:z()) > extnav_innov_thresh)
+
+  -- get rangefinder distance
+  local rngfnd_distance_m = 0
+  if rangefinder:has_data_orient(rangefinder_rotation) then
+    rngfnd_distance_m = rangefinder:distance_cm_orient(rangefinder_rotation) * 0.01
+  end
+  local rngfnd_over_threshold = (rngfnd_distance_m == 0) or (rngfnd_distance_m > rangefinder_thresh_dist)
+
+  -- NpnGPS is usable if extnav innovations are good or rangefinder distance is short (for optical flow)
+  local nongps_usable = (not extnav_over_threshold) or (not rngfnd_over_threshold)
+
+  -- debug.debug
+  debug_counter = debug_counter + 1
+  if (debug_counter > 30) then
+    debug_counter = 0
+    --gcs:send_text(0, string.format("gpsu:%s ot:%s extot:%s rngot:%s nongpsu:%s", tostring(gps_usable), tostring(gps_over_threshold), tostring(extnav_over_threshold), tostring(rngfnd_over_threshold), tostring(nongps_usable)))
+  end
```

Comment:
Can all be commented rather than just he send_text

Reply:
debug removed, txs!